support/ci/install-nightly-rustfmt.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# This script can be run standalone, but it can also be sourced from rustfmt.sh.
+# If it's sourced, then $rustup is already defined for us. If it's not, we need
+# to make sure we have it.
+: "${rustup:=$(command -v rustup)}"
+
+# Due to the nature of nightly rust, sometimes changes will break rustfmt's
+# usage of rustc. If this happens, nightly rust won't include rustfmt,
+# and we need to automatically fall back to a version that does include it.
+# Note that we begin with 1 day ago, since nightly packages can sometimes not
+# exist when this script runs if we leave it at 0.
+max_days=90
+
+for days_ago in $(seq 1 1 $max_days)
+do
+  date=$(date -d "$days_ago days ago" +%Y-%m-%d)
+  toolchain="nightly-$date"
+  echo "Installing rust $toolchain"
+  sudo -E "$rustup" toolchain install "$toolchain"
+
+  if sudo -E "$rustup" component add --toolchain "$toolchain" rustfmt; then
+    if [ "${BASH_SOURCE[0]}" != "${0}" ]; then
+      # If we made it here, this script is being sourced from rustfmt.sh, so do the check
```

Comment:
This is a little surprising to see in a file named `install-nightly-rustfmt.sh`. Can things be refactored such that the separation of concerns is more clear?